### DISCUSSION TITLE
the component table can cause memory overflow

### DISCUSSION THREAD

**Discussion by chenjinhuizz** [element-plus/element-plus#14975] [Q&A]
When I read the element plus doc, I check component table, then check other componet, and then check table back again. I do the same operation several times. The memory used rise from 30MB to 200MB. If I use lots of components table in my app, it will cause memory overflow and my app run out.

**Comment by emojiiii** [comment]
this problem is likely to be caused by 'Vue'

**Comment by chenjinhuizz** [comment]
But the element ui base on vue2 use very few memory, about 20MB. So this problem is caused by Vue3?
I check the iview doc base on vue2. It cause the memory overflow too. 
I am a java coder. I am not good at js. So I just guess this is caused by not destroying the dom when router change.
Due to this problem, I still stay with element ui.

[a comment by an automated account is not shown]

**Comment by caozhong1996** [comment]
今天看了一下你提到的问题，Vue3确实存在一些内存泄漏的情况，例如内部绑定点击事件造成了闭包导致无法回收，但是并不严重，晚些时候会把今天发现的问题給Vue提交PR。
PS: 我目前在大数据方向的组工作，也用了vue3 + elementplus，拿你提到的表单举例，实践证明即使在数据量巨大的情况下也没有太大的影响。前端并不像后端那样对于内存敏感，尤其是数据量小的情况下可以放心使用，现存的问题也都会慢慢修复的。

**Comment by AK47-dadada** [comment]
> 今天看了一下你提到的问题，Vue3确实存在一些内存泄漏的情况，例如内部绑定点击事件造成了闭包导致无法回收，但是并不严重，晚些时候会把今天发现的问题給Vue提交PR。 PS: 我目前在大数据方向的组工作，也用了vue3 + elementplus，拿你提到的表单举例，实践证明即使在数据量巨大的情况下也没有太大的影响。前端并不像后端那样对于内存敏感，尤其是数据量小的情况下可以放心使用，现存的问题也都会慢慢修复的。

坐等更新啊，一周又一周.

**Comment by chenjinhuizz** [comment]
**问题重现方法**
打开element plus官方文档的时候, 内存使用率只有6MB
点击Table的菜单, 然后切换到其它菜单, 再切回Table菜单.
重复以上操作10次, 内存大概会爬升到80MB.
如果依次点击【Data数据展示】的所有菜单, 内存大概会攀升到250MB+, 而且不会回收, 随着内存占用会越来越大.
**实际开发中会出现的场景**
这个问题在开发运营类型的系统时, 特别突出.运营类的系统会有大量的table, 分布在不同的菜单. 使用过程会在不同的菜单间来回切换. 当内存使用到300MB~400MB的时候, 页面就会出现白屏. 需要刷新清空内存. 


> Hello @chenjinhuizz. Please provide an online reproduction demo by clicking this [link](https://codepen.io/iamkun/pen/YzWMaVr) or a minimal GitHub repository.你好 @chenjinhuizz, 请提供一个可复现问题的链接以便于我们帮你排查问题。可以通过点击 [此处](https://codepen.io/iamkun/pen/YzWMaVr) 创建或者提供一个最小化的 GitHub 仓库。



> 今天看了一下你提到的问题，Vue3确实存在一些内存泄漏的情况，例如内部绑定点击事件造成了闭包导致无法回收，但是并不严重，晚些时候会把今天发现的问题給Vue提交PR。 PS: 我目前在大数据方向的组工作，也用了vue3 + elementplus，拿你提到的表单举例，实践证明即使在数据量巨大的情况下也没有太大的影响。前端并不像后端那样对于内存敏感，尤其是数据量小的情况下可以放心使用，现存的问题也都会慢慢修复的。

**Comment by FlyBirdF** [comment]
我也遇到了内存泄漏的问题，我使用的Vue3+element-plus，数据是从后端websocket中不停的推送到前端的el-table中，因为数据是每秒钟刷新的，运行一段时间后会导致内存越来越大，直到网页因为内存问题崩溃。用chrome dev内存工具检查的时候js heap大小会逐步增大但是每个ticker增大的幅度不大，memory是一直变大的并且以肉眼可见的速度增长能到达每个ticker净增长1M左右的内存空间。

**Comment by Wincy94** [comment]
项目使用了自定义插槽开发可编辑表格时，因为绑定了很多表单控件，这个问题变得很严重。请问有好的解决方案吗？

**Comment by emojiiii** [comment]
Seems to have found the problem.

**Comment by star9923** [comment]
el-table 添加fixed 在不同版本会有不同的问题，添加fixed会导致页面崩溃

**Comment by chenxch** [comment]
@emojiiii Is there a sequel to this?

**Comment by ajesses** [comment]
这个问题解决了吗？
目前版本：vue 3.2.39，element-plus 2.2.17
内存还是在增加，不回收

**Comment by YihooZero** [comment]
> 我也遇到了内存泄漏的问题，我使用的Vue3+element-plus，数据是从后端websocket中不停的推送到前端的el-table中，因为数据是每秒钟刷新的，运行一段时间后会导致内存越来越大，直到网页因为内存问题崩溃。用chrome dev内存工具检查的时候js heap大小会逐步增大但是每个ticker增大的幅度不大，memory是一直变大的并且以肉眼可见的速度增长能到达每个ticker净增长1M左右的内存空间。

@FlyBirdF 你好，我这边遇到了类似的问题，请问你这边最后怎么解决的，谢谢

**Comment by FlyBirdF** [comment]
> > 我也遇到了内存泄漏的问题，我使用的Vue3+element-plus，数据是从后端websocket中不停的推送到前端的el-table中，因为数据是每秒钟刷新的，运行一段时间后会导致内存越来越大，直到网页因为内存问题崩溃。用chrome dev内存工具检查的时候js heap大小会逐步增大但是每个ticker增大的幅度不大，memory是一直变大的并且以肉眼可见的速度增长能到达每个ticker净增长1M左右的内存空间。
> 
> @FlyBirdF 你好，我这边遇到了类似的问题，请问你这边最后怎么解决的，谢谢
@YihooZero 你这个现象是在开发环境中遇到的么？我是尝试打包编译项目到uat环境中，然后就没有上述在开发环境中出现的比较严重的内存泄露问题了。你也可以编译一下试试看看。

**Comment by YihooZero** [comment]
> > > 我也遇到了内存泄漏的问题，我使用的Vue3+element-plus，数据是从后端websocket中不停的推送到前端的el-table中，因为数据是每秒钟刷新的，运行一段时间后会导致内存越来越大，直到网页因为内存问题崩溃。用chrome dev内存工具检查的时候js heap大小会逐步增大但是每个ticker增大的幅度不大，memory是一直变大的并且以肉眼可见的速度增长能到达每个ticker净增长1M左右的内存空间。
> > 
> > 
> > @FlyBirdF 你好，我这边遇到了类似的问题，请问你这边最后怎么解决的，谢谢
> > @YihooZero 你这个现象是在开发环境中遇到的么？我是尝试打包编译项目到uat环境中，然后就没有上述在开发环境中出现的比较严重的内存泄露问题了。你也可以编译一下试试看看。

@FlyBirdF 目前是在开发环境中的，我把el-table换成原生的table渲染list数据，还是会出现内存不断增长的情况（我这边有个300条的list数据，其中一个字段是音量大小，每隔2s会改变刷新这个字段，内存就会越来越大）

**Comment by songyx95** [comment]
内存泄漏的问题还存在，头疼了。我现在项目还是会时不时的崩溃了。vue：3.2.41，element-plus：2.2.19

**Comment by chenxch** [comment]
> 内存泄漏的问题还存在，头疼了。我现在项目还是会时不时的崩溃了。vue：3.2.41，element-plus：2.2.19

方便提供一个demo么

**Comment by chenjinhuizz** [comment]
> > 内存泄漏的问题还存在，头疼了。我现在项目还是会时不时的崩溃了。vue：3.2.41，element-plus：2.2.19
> 
> 方便提供一个demo么

打开element plus文档, 反复点击Table, Virtualized Tree. 点个十来次, 按F12, 查看memory, 可以飙升到1GB以上. 一段时间不操作, 大约可以稳定在300+M
但这只是文档静态demo. 实际在生产中, 因为列字段多, 数据需要不断从后端取回重新渲染. 再配合业务需求, field自定义slot. 重数据的后台管理系统, 几乎一段时间就需要刷新一次页面清理内存.

我目前, 还停留在vue2+elment.

**Comment by songyx95** [comment]
> > 内存泄漏的问题还存在，头疼了。我现在项目还是会时不时的崩溃了。vue：3.2.41，element-plus：2.2.19
> 
> 方便提供一个demo么

上面已经有朋友给出例子了，我现在项目大量用到table和dialog。然后里面又有自定义组件或者dialog里有表单，现在的情况是时不时的出现，也没有一定的规律。有时候就停留在页面不做任何操作都会网页崩溃了out of memory

**Comment by chenxch** [comment]
好的，如果你们有目前大概定位到的相关信息也可以提供一下。

**Comment by ganl** [comment]
+1

**Comment by Mario34** [comment]
https://github.com/vuejs/core/issues/8790 相关问题

**Comment by xiaoxian521** [comment]
升级到 `element-plus@2.4.3` 再试试
如果还有问题 请提供个最小复现仓库 请不要用`https://github.com/element-plus/element-plus-playground`或者文档来复现，因为它们本身会影响到要测试的组件，我们需要个最小复现仓库（如丢个`github`链接）然后`README.md`备注清楚如何复现，我们将尽力调查验证并解决

**Reply by star-hs** [reply]
升级到2.4.4，仍未解决。是el-table-column插槽的问题，但具体为什么不清楚

**Reply by Happy-DJ** [reply]
element-plus:2.5.1,vue 3.2.47 el-table-column还是会导致内存溢出

**Reply by Alone-Michael** [reply]
和vue3 的版本有关系吗？还是说只需要升级element-plus 的版本

**Reply by kiss-keray** [reply]
> 升级到`element-plus@2.4.3`再尝试 如果还有问题请提供一个最小恢复仓库请不要用`https://github.com/element-plus/element-plus-playground`或者文档来恢复，因为它们本身会影响到要测试的组件，需要我们一个最小恢复仓库（如丢了一个`github`链接）然后`README.md`备注清楚如何详细说明，我们将尽力调查验证并解决

现在是element-plus:2.8.3 vue:3.5.6  表格每行有2个el-input  过几秒刷新一次table  还是会内存溢出

**Comment by Alone-Michael** [comment]
我也遇见内存溢出，测试在一个页面重复打开弹框，弹框中有form表单，重复打开多次后，内存飙升，然后页面崩溃

**Comment by elarbor** [comment]
表格就该用一个组件，就叫`<el-table>` 。现在是两个[`<el-table>` ](https://element-plus.org/zh-CN/component/table.html)和[`<el-table-v2>`](https://element-plus.org/zh-CN/component/table-v2.html) 。

用type区分更好一点：
`<el-table type="default">` 就是原来的`<el-table>`组件
`<el-table type="virtualized">`  就是原来的`<el-table-v2>` 组件
`<el-table type="第三种">`  就是以后的`<第三种>` 组件。    ......

**Comment by changxuexu** [comment]
这个内存溢出到底是啥导致的，我最近使用这个版本也是发现很多游离的节点。。。。

**Comment by geekli** [comment]
vue 从`3.5.3` 升级到` 3.5.13` el-table 内存泄露问题解决了
element-plus版本一直是`2.8.1`
